app/src/main/java/com/keylesspalace/tusky/components/report/adapter/StatusViewHolder.kt

```diff
@@ -0,0 +1,166 @@
+        setCreatedAt(status.createdAt)
+    }
+
+    private fun updateTextView() {
+        status()?.let { status ->
+            setupCollapsedState(status.isCollapsible(), viewState.isCollapsed(status.id, true),
+                    viewState.isContentShow(status.id, status.sensitive), status.spoilerText)
+
+            if (status.spoilerText.isBlank()) {
+                setTextVisible(true, status.content, status.mentions, status.emojis, adapterHandler)
+                itemView.statusContentWarningButton.visibility = View.GONE
+                itemView.statusContentWarningDescription.visibility = View.GONE
+            } else {
+                val emojiSpoiler = CustomEmojiHelper.emojifyString(status.spoilerText, status.emojis, itemView.statusContentWarningDescription)
+                itemView.statusContentWarningDescription.text = emojiSpoiler
+                itemView.statusContentWarningDescription.visibility = View.VISIBLE
+                itemView.statusContentWarningButton.visibility = View.VISIBLE
+                itemView.statusContentWarningButton.isChecked = viewState.isContentShow(status.id, true)
+                itemView.statusContentWarningButton.setOnCheckedChangeListener { _, isViewChecked ->
+                    status()?.let { status ->
+                        itemView.statusContentWarningDescription.invalidate()
+                        viewState.setContentShow(status.id, isViewChecked)
+                        setTextVisible(isViewChecked, status.content, status.mentions, status.emojis, adapterHandler)
+                    }
+                }
+                setTextVisible(viewState.isContentShow(status.id, true), status.content, status.mentions, status.emojis, adapterHandler)
+            }
+        }
+    }
+
+
+    private fun setTextVisible(expanded: Boolean,
+                               content: Spanned,
+                               mentions: Array<Status.Mention>?,
+                               emojis: List<Emoji>,
+                               listener: LinkListener) {
+        if (expanded) {
+            val emojifiedText = CustomEmojiHelper.emojifyText(content, emojis, itemView.statusContent)
+            LinkHelper.setClickableText(itemView.statusContent, emojifiedText, mentions, listener)
+        } else {
+            LinkHelper.setClickableMentions(itemView.statusContent, mentions, listener)
+        }
+        if (itemView.statusContent.text.isNullOrBlank()) {
+            itemView.statusContent.visibility = View.GONE
+        } else {
+            itemView.statusContent.visibility = View.VISIBLE
+        }
+    }
+
+    private fun setCreatedAt(createdAt: Date?) {
+        if (useAbsoluteTime) {
+            itemView.timestampInfo.text = statusViewHelper.getAbsoluteTime(createdAt)
+        } else {
+            itemView.timestampInfo.text = if (createdAt != null) {
+                val then = createdAt.time
+                val now = Date().time
```

Comment:
`System.currentTimeMillis()` is slighty faster because it does not create an object

Reply:
Agree. Done